src/core/cgroup.c

```diff
@@ -1381,7 +1407,9 @@ static void cgroup_context_apply(
          * we couldn't even write to them if we wanted to). */
         if ((apply_mask & CGROUP_MASK_CPU) && !is_local_root) {
 
-                if (cg_all_unified() > 0) {
+                if (cgroup_context_is_cpu_idle(c))
+                        cgroup_apply_unified_cpu_idle(u, true);
```

Comment:
how does the idle field actually work? if i want to switch a running service from idle mode to non-idle mode, wouldn't i have to first write "0" into the idle cgroupfs attr file, and then fix the weight value? Or does setting the weight value suffice to turn off idle mode again?

it appears to me, you kinda have to write the idle field always to make this robust, no?

Reply:
> how does the idle field actually work? 

If we switch service from idle mode to non-idle mode, we have to set idle attr to `0` and then give a weight, and if we need to switch it from non-idle to idle, we only need set idle attr to `1` and mustn't give it a weight.

> it appears to me, you kinda have to write the idle field always to make this robust, no?

*I am not sure I got it* 
yes, we should always write this idle field. we can't modify the weight of an idle service, so write this idle field to `0` before modifying its weight. do you means we should do: 
```
if (cgroup_context_is_cpu_idle(c))
  cgroup_apply_unified_cpu_idle(u, true);
else {
  cgroup_apply_unified_cpu_idle(u, false);
  // modify some weight
}

Reply:
yes, something like that